packaging/curated-cloud-installer.spec

```diff
@@ -0,0 +1,137 @@
+mkdir -p %{buildroot}%{data_dir}
+
+mv vendor %{buildroot}%{lib_dir}
+mv db %{buildroot}%{data_dir}
+mv tmp %{buildroot}%{data_dir}
+mv log %{buildroot}%{data_dir}
+cp -ar . %{buildroot}%{app_dir}
+
+ln -s %{data_dir}/tmp %{buildroot}%{app_dir}/tmp
+ln -s %{data_dir}/db %{buildroot}%{app_dir}/db
+ln -s %{data_dir}/log %{buildroot}%{app_dir}/db
+
+# systemd
+mkdir -p %{buildroot}%{_unitdir}
+install -m 444 server-configs/systemd/curated-cloud-installer.service %{buildroot}%{_unitdir}
+
+# nginx
+install -D -m 644 server-configs/nginx/curated-cloud-installer.conf %{buildroot}%{_sysconfdir}/nginx/vhosts.d/curated-cloud-installer.conf
+
+# set relocated destination for gems
+sed -i -e '/BUNDLE_PATH: .*/cBUNDLE_PATH: "\/usr\/lib64\/curated-cloud-installer\/vendor\/bundle\/"' \
+    -e 's/^BUNDLE_JOBS: .*/BUNDLE_JOBS: "1"/' \
+    %{buildroot}%{app_dir}/.bundle/config
+
+# cleanup of /usr/bin/env commands
+grep -rl '\/usr\/bin\/env ruby' %{buildroot}%{lib_dir}/vendor/bundle/ruby | xargs \
+    sed -i -e 's@\/usr\/bin\/env ruby.%{ruby_version}@\/usr\/bin\/ruby\.%{ruby_version}@g' \
+    -e 's@\/usr\/bin\/env ruby@\/usr\/bin\/ruby\.%{ruby_version}@g'
+grep -rl '\/usr\/bin\/env bash' %{buildroot}%{lib_dir}/vendor/bundle/ruby | xargs \
+    sed -i -e 's@\/usr\/bin\/env bash@\/bin\/bash@g'
+
+# cleanup unneeded files
+find %{buildroot}%{lib_dir} "(" -name "*.c" -o -name "*.h" -o -name .keep ")" -delete
+find %{buildroot}%{app_dir} -name .keep -delete
+find %{buildroot}%{data_dir} -name .keep -delete
+rm %{buildroot}%{app_dir}/README.md
+rm %{buildroot}%{app_dir}/LICENSE
+rm -r  %{buildroot}%{lib_dir}/vendor/bundle/ruby/[23].*.0/cache
+rm -rf %{buildroot}%{lib_dir}/vendor/cache
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/doc
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/examples
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/samples
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/test
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/ports
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/ext
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/bin
+rm -rf %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/spec
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/.gitignore
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/*/*/*/.gitignore
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/extensions/*/*/*/gem_make.out
+rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/extensions/*/*/*/mkmf.log
+rm -rf %{buildroot}%{app_dir}/server-configs
+
+%fdupes %{buildroot}/%{app_dir}
+%fdupes %{buildroot}/%{lib_dir}
+
+# drop custom rpath from native gems
+chrpath -d %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/nokogiri-*/lib/nokogiri/*/nokogiri.so
+
+%files
+%attr(-,root,root) %{app_dir}
+%doc README.md LICENSE
+%{lib_dir}
+%attr(-,root,root) %{data_dir}
+%{_unitdir}/curated-cloud-installer.service
+%dir %{_sysconfdir}/nginx
+%dir %{_sysconfdir}/nginx/vhosts.d
```

Comment:
These two directories are owned by the `nginx` package, so you should not try to own them in your own package but rather add a dependency to `nginx`.

Also, please be aware that the configuration files for `nginx` don't seem to be necessarily be placed below `%{_sysconfdir}`, see: https://build.opensuse.org/package/view_file/server:http/nginx/nginx.spec?expand=1

Reply:
I don't know why the _nginx_ package goes to great lengths to abstract it, but it is as described. the `nginx-macros` package[1] defines the macro as:

    %ngx_conf_dir   %{_sysconfdir}/nginx
    
[1] Line 4: https://build.suse.de/package/view_file/SUSE:Factory:Head/nginx-macros/nginx.macros?expand=1

Reply:
WRT ownership, `nginx` _is_ required, I get this if I don't include the dirs:

```
BUILD FAILED!
...
[  118s] curated-cloud-installer-0.0.1-150300.2.1.x86_64.rpm: directories not owned by a package:
[  118s]  - /etc/nginx
[  118s]  - /etc/nginx/vhosts.d
```

Reply:
Okay, I got it to work by adding `BuildRequires:  nginx`